### PR TITLE
[c++] Rename `SOMAArrayReader` to `SOMAArray`

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue and/or context:**

#1282

This is one of the several changes made in #1259 that will be broken off into smaller PRs.

**Changes:**

Rename `SOMAArrayReader` to `SOMAArray`: In an upcoming PR, will add `SOMAGroup`

### CODE REVIEW

File: apis/r/R/RcppExports.R [hunk 1]
```diff
@@ -3,31 +3,17 @@
 
 #' Read SOMA Data From a Given URI
 #'
-#' This functions access a given SOMA URI and returns a complete data.frame. It does
-#' not iterate; if your data is large than the initial read size consider the \code{sr_*}
-#' functions.
+#' This functions access a given SOMA URI and returns a complete data.frame. It
+NULL
```
[the remark this answers is not in the document]
Comment (johnkerl) [reply apis/r/R/RcppExports.R:7]: OTOH IIRC this is Rcpp autogen, so, nvm

File: apis/r/src/rutilities.cpp [hunk 2]
```diff
@@ -1,146 +1,211 @@
...
-#include "rutilities.h"         // local declarations
+#include "rutilities.h"  // local declarations
 
 namespace tdbs = tiledbsoma;
 
-void apply_dim_points(tdbs::SOMAArrayReader *sr,
-                      std::unordered_map<std::string, std::shared_ptr<tiledb::Dimension>>& name2dim,
-                      Rcpp::List lst) {
+void apply_dim_points(
+    tdbs::SOMAArray* sr,
```
Comment (johnkerl) [inline apis/r/src/rutilities.cpp:14]: was this `clang-format` -- or something else? just curious why these are appearing

File: apis/r/src/Makevars.in [hunk 3]
```diff
@@ -1,7 +1,7 @@
 CXX_STD = CXX17
 
 ## We need the TileDB Headers, and for macOS aka Darwin need to set minimum version 10.14 for macOS
-PKG_CPPFLAGS = -I. -I../inst/include/ @tiledb_include@ -DR_BUILD @cxx17_macos@ -Iinclude
+PKG_CPPFLAGS = -I. -I../inst/include/ -Ilibtiledbsoma/external/include @tiledb_include@ -DR_BUILD @cxx17_macos@ -Iinclude
```
Comment (eddelbuettel) [inline apis/r/src/Makevars.in:4]: There is a large parallel pending change that removes all this when we switch to a shared libtiledbsoma library.

Bigger picture the -I location should not be hardwired by learned in `configure` probably from `pkg-config`

File: apis/r/src/rinterface.cpp [hunk 4]
```diff
@@ -8,104 +8,113 @@
...
 Rcpp::XPtr<ArrowSchema> schema_owning_xptr(void) {
-  struct ArrowSchema* schema = (struct ArrowSchema*)ArrowMalloc(sizeof(struct ArrowSchema));
-  if (schema == NULL) Rcpp::stop("Failed to allocate ArrowSchema");
-  schema->release = NULL;
-  Rcpp::XPtr<ArrowSchema> schema_xptr = make_xptr(schema, false);
-  return schema_xptr;
+    struct ArrowSchema* schema = (struct ArrowSchema*)ArrowMalloc(
+        sizeof(struct ArrowSchema));
+    if (schema == NULL)
+        Rcpp::stop("Failed to allocate ArrowSchema");
```
Comment (eddelbuettel) [inline apis/r/src/rinterface.cpp:23]: I would really really prefer if we stuck to existing coding convention which are to NOT use clang-format here.

**Review by johnkerl — COMMENTED** [review]
LGTM although I defer to @Shelnutt2 for final approval

Thanks for working on this! Epic quantity of work happening with amazing speed

This PR is way too big, and hard to wrap one's head around, and with PRs of this size it's easy for any reviewer to miss subtle things among all the things going on.

Most of the line-count here appears to be whitespacing/formatting, between C++ source and `CMakeLists.txt` changes.

I won't push back on this particular PR, but in future, I do ask that you do whitespace-only PRs separately. A many-thousand-line whitespace-only PR is a no-brainer for anyone to approve. Then, a separate, substantive PR with smaller line-count will give everyone the opportunity to focus on the high-value coding work per se.

**Review by eddelbuettel — COMMENTED** [review]
There are a very high number of affected files here, in part simply because a formatter was run over the code.  

I was wondering if we could avoid that and make the PR more minimal?

**Comment by nguyenv** [comment]
I'll turn off auto-formatting with clang-format and also redo the PR without the formatting changes.

**Comment by johnkerl** [comment]
> I'll turn off auto-formatting with clang-format and also redo the PR without the formatting changes.

@nguyenv how about:

* branch cleanly off `main`
* run `clang-format`
* put up a PR _for that only_
* rebase your other PRs on top of that

?

**Comment by eddelbuettel** [comment]
Alternate:

- run `clang-format` only on `libtiledbsoma/`

**Review by eddelbuettel — APPROVED** [review]
Much cleaner and more manageable!

**Review by johnkerl — APPROVED** [review]
🚢

**Pull request merged**